rasa/core/policies/intent_ted_policy.py

```diff
@@ -482,6 +484,25 @@ def predict_action_probabilities(
             action_metadata=self._collect_action_metadata(domain, similarities),
         )
 
+    @staticmethod
+    def _is_queried_after_action_executed(tracker: DialogueStateTracker) -> bool:
+        """Checks if query was made after an `ActionExecuted` event.
+
+        Should check if there is an `ActionExecuted` after the last `UserUttered`.
+
+        Returns:
+            Whether the policy was queried after and `ActionExecuted`.
+        """
+        applied_events = tracker.applied_events()
+
+        found_action = False
+        for event in reversed(applied_events):
+            if isinstance(event, ActionExecuted):
+                found_action = True
```

Comment:
Couldn't you also just return here?

Reply:
Ahh, now I see what you both meant 🙈 . In my head I was always thinking of trying to find if `ActionExecuted` occured after a `UserUttered` only. But i see now why it can just be simplified to checking which occurred first from the back of the list of the events - `ActionExecuted` or `UserUttered`.
I'll change it on monday morning.